dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java

```diff
@@ -338,25 +338,29 @@ private View createView(LayoutInflater inflater) {
         holder.setBackgroundColor(backgroundColorResourceId);
         View view = holder.getView(inflater, rootView);
 
+        if (holder instanceof ViewHolder) {
+            loopViewsRecursively(view);
+        }
+
         loopViewsRecursively(headerView);
         holder.addHeader(headerView);
 
         loopViewsRecursively(footerView);
         holder.addFooter(footerView);
 
-        if (adapter != null) {
-            holder.setAdapter(adapter);
-        }
+        if (adapter != null && holder instanceof HolderAdapter) {
+            ((HolderAdapter) holder).setAdapter(adapter);
```

Comment:
local variable would be nice here. We can avoid one extra casting
<pre>
HolderAdapter holderAdapter = (HolderAdapter) adapter;
</pre>